new(content/en/blog): 0.29.0 release (#457)
falcosecurity/falco-website: content/en/blog/falco-0-29-0.md

---
title: Falco 0.29.0
date: 2021-06-21
author: Massimiliano Giovagnoli
slug: falco-0-29-0
---

Today we announce the summer release of Falco 0.29.0 🌱

> This is a minor version which brings new features and fixes!

Reviewer: I wouldn't say this is "minor" :)

## Novelties 🆕

Let's now review some of the new things Falco 0.29.0 brings.

> ### New libraries repostiory!

Reviewer: ```suggestion
### New libraries repository!
```

> First of all, as per [this proposal](https://github.com/falcosecurity/falco/blob/master/proposals/20210119-libraries-contribution.md), the repo [falcosecurity/libs](https://github.com/falcosecurity/libs) is the new home for [`libscap`](https://github.com/falcosecurity/libs/tree/master/userspace/libscap), [`libsinsp`](https://github.com/falcosecurity/libs/tree/master/userspace/libsinsp), and the Falco [drivers](https://github.com/falcosecurity/libs/tree/master/driver).

Reviewer: ```suggestion
As per [this proposal](https://github.com/falcosecurity/falco/blob/master/proposals/20210119-libraries-contribution.md) - and as many of you probably already know - the repo [falcosecurity/libs](https://github.com/falcosecurity/libs) is the new home for [`libscap`](https://github.com/falcosecurity/libs/tree/master/userspace/libscap), [`libsinsp`](https://github.com/falcosecurity/libs/tree/master/userspace/libsinsp), and the Falco [drivers](https://github.com/falcosecurity/libs/tree/master/driver).
```

> With this release the building system is updated to point to the [new location](https://download.falco.org/?prefix=driver/13ec67ebd23417273275296813066e07cb85bc91/) and also the [driver version](https://download.falco.org/?prefix=driver/13ec67ebd23417273275296813066e07cb85bc91/) is updated.

Reviewer: ```suggestion
With this release, also the last missing piece of the libs contribution is done: the building system is now updated to point to the [new location](https://download.falco.org/?prefix=driver/17f5df52a7d9ed6bb12d3b1768460def8439936d/) and also the [driver version](https://download.falco.org/?prefix=driver/17f5df52a7d9ed6bb12d3b1768460def8439936d/) is updated.
```

### New libs version!

The update to the [drivers](https://github.com/falcosecurity/libs/tree/master/driver) version [17f5d](https://github.com/falcosecurity/libs) brings new features/fixes:

> - [support](https://github.com/falcosecurity/libs/pull/50) to trace the [userfaultd](https://www.kernel.org/doc/html/latest/admin-guide/mm/userfaultfd.html) system call

Reviewer: ```suggestion
- [support](https://github.com/falcosecurity/libs/pull/50) for tracing the [userfaultd](https://www.kernel.org/doc/html/latest/admin-guide/mm/userfaultfd.html) system calls
```

> - [improvement](https://github.com/falcosecurity/libs/pull/32) in [`libsinsp`](https://github.com/falcosecurity/libs/tree/master/userspace/libsinsp) on the way how pod resources limits and pod IP are gathered from container runtime

Reviewer: ```suggestion
- [improvements](https://github.com/falcosecurity/libs/pull/32) to how [`libsinsp`](https://github.com/falcosecurity/libs/tree/master/userspace/libsinsp) gathers Kubernetes pod resources limits and pod IP from the container runtime
```

> - [improvement](https://github.com/falcosecurity/libs/pull/15) in [`libsinsp`](https://github.com/falcosecurity/libs/tree/master/userspace/libsinsp) on pod metadata and namespace retrievial for large cluster scenarios, by getting them directly from container labels and bypassing the API server

Reviewer 2: ```suggestion
- [improvement](https://github.com/falcosecurity/libs/pull/15) in [`libsinsp`](https://github.com/falcosecurity/libs/tree/master/userspace/libsinsp) on pod metadata and namespace retrieval for large cluster scenarios, by getting them directly from container labels which is more efficient and use the K8s API server as a fallback
```

Reviewer 2: IMO, it worth also mentioning:
- https://github.com/falcosecurity/libs/pull/22
- https://github.com/falcosecurity/libs/pull/42

Reviewer: agree

### Improvements on building system

Finally, it introduces necessary adaptations and improvements to make the Falco building system work with changes recently introduced in [libs](https://github.com/falcosecurity/libs) `CMakefile`s (in particular by PRs [#23](https://github.com/falcosecurity/libs/pull/23) and [#30](https://github.com/falcosecurity/libs/pull/30)).

### Updated rules

As usual, we keep improving the existing rules and we added new ones, like [removing false positives](https://github.com/falcosecurity/falco/pull/1665) when detecting non-sudo and non-root setuid calls.

Other false positives has been removed by [ignoring](https://github.com/falcosecurity/falco/pull/1659) additional known Kubernetes service account when watching for service accounts creted in `kube-system` namespace.

> Improvments have been made also for anti-miner detection, by [adding additional domains](https://github.com/falcosecurity/falco/pull/1676) to be detected.

Reviewer 2: ```suggestion
Improvements have been made also for anti-miner detection, by [adding additional domains](https://github.com/falcosecurity/falco/pull/1676) to be detected.
```
